### DISCUSSION TITLE
Is there a cordova plugin that exposes navigator.mediaDevices ?

### DISCUSSION THREAD

**Discussion by ghevge** [apache/cordova#398] [Q&A]
Hi,
I'm trying to get access to the camera using the navigator.mediaDevices.getUserMedia({video: true}) approach, but so far, I had no luck in Cordova. I was trying with cordova-plugin-media-capture & cordova-plugin-camera.

This approach works flawless in chrome though.

Is there a way to make it run in Cordova? 

I am force to stick with this approach, as one of the 3rd party libraries my app is using, is using this method. 

Thanks.

**Comment by BigBalli** [comment]
What do you mean by "had no luck"? Does it crash? errors shown?
Something boilerplate like this seems to work fine.

```
const videoContainer = document.getElementById('video-container');

navigator.mediaDevices.getUserMedia({ video: true })
  .then(stream => {
    const videoElement = document.createElement('video');
    videoElement.srcObject = stream;
    videoElement.autoplay = true;
    videoElement.muted = true; // Mute to avoid potential feedback
    videoContainer.appendChild(videoElement);
  })
  .catch(error => {
    console.error('Error accessing webcam:', error);
  });

```

**Comment by ghevge** [comment]
It was a permission problem. It started working when I've added these entries in config.xml

```
<platform name="android">
	    <config-file target="AndroidManifest.xml" parent="/manifest">
	        <uses-permission android:name="android.permission.CAMERA"/>
	    </config-file>
	</platform>
```

Before I did this, the navigator.mediaDevices.getUserMedia was always returning 0 devices.
